### PR TITLE
Update to1.3.0

### LINKED ISSUE #2039: Compatibility with KSP 1.3
Now that it's a full release and not pre-release, let's have an issue to handle the changes needed to make things work with KSP 1.3.

The remit of this issue is just making things work, not about using any new features or enhancements or anything like that.

### PR DESCRIPTION
**Pull request by pipakin**
Absolute minimum changes necessary to get it to run in 1.3.0. I've been using it all day without issue.

Fixes #2039

### CODE REVIEW

**Comment by pipakin** [comment]
We need to get the 1.3.0 dlls in the KSP_LIB project so this can succeed the Travis build, then the devs can merge this change.

**Comment by pipakin** [comment]
If you are impatient, you'll have to pull down my branch, copy the dlls it references into the resources folder, and build it in visual studio.  Once you've done that you can copy the Resources/GameData folder to your ksp gamedata folder and you'll have a working mod in 1.3.0! :)

**Comment by sebseb7** [comment]
can someone please provide a binary?

thanx

**Comment by scorpianz1525** [comment]
Is there an issue with this then? I've been looking to just get the update just to run an old script I have.

**Comment by pipakin** [comment]
[kOS.zip](https://github.com/KSP-KOS/KOS/files/1065941/kOS.zip)
Here's my binaries, I have no idea if they will work for anyone other than me so use at your own risk. As soon as they get the build process updated for 1.3.0 you should be able to use the official build once this gets merged in.

**Comment by scorpianz1525** [comment]
Thanks!

**Comment by pipakin** [comment]
Added a PR to https://github.com/KSP-KOS/KSP_LIB that should update the build process so we can get the build passing. PR for reference: https://github.com/KSP-KOS/KSP_LIB/pull/3

**Comment by scorpianz1525** [comment]
Are the 'labels' working?

**Comment by scorpianz1525** [comment]
As in the right click set label thing. That seems to be missing?

**Comment by Dunbaratu** [comment]
I promise I do plan to get to this soon.  We were stalled for a while because we had a rule that no pull requests can be merged without a peer review (someone else must do your merges for you), and hvacengi got swamped at work with no free time for kOS.  Since the active dev team was down to just us two, that stalled everything as every pull request required at least 2 people involved.

I've gotten temporary permission to violate our own rules and make merges without reviews so we can break the deadlock and get a release out.  (I'm a little nervous about this but there's not much we can do about it).

The plan is:

Step 1 - get a last release of kOS for KSP 1.2.2 out for use by all those who will be stuck on 1.2.2 for a while.
Step 2 - get a release of kOS for KSP 1.3 out based on Step 1 plus the 1.3 changes.

I'm still back at Step 1, but with the permission to break the peer review rule, I should be able to finally make progress.

**Comment by pipakin** [comment]
Hey no worries, I know how that can be. :)  Let me know if there's anything I can do to help. Also note this was the very first time I ever modified a KSP mod's code, so let me know if I did anything horrible.

**Comment by scorpianz1525** [comment]
Cool, could you keep us updated on progress? :)

**Comment by fiktor** [comment]
Using kOS.zip attached above in KSP 1.3.0.1804 (on Ubuntu 16.04 64bit), kOS mostly works: I can get kOS terminal and can execute most of the commands. One error I get is when trying to run the following inside kOS terminal:
PRINT SHIP:ORBIT:EPOCH.

It gives this error:
Get Suffix 'EPOCH' not found on object ORBIT of ship name.

This is not a big deal, just a notice: there are clearly workarounds (e.g. one can just get SHIP:ORBIT:TRUEANOMALY and assume that epoch is TIME).

**Comment by Dunbaratu** [comment]
EPOCH was just added two days ago.  This PR was created before then.  updating the PR to the latest ``develop`` will fix this.

**Comment by scorpianz1525** [comment]
Like I said above the name tags don't work either

**Comment by Dunbaratu** [comment]
name tags depend on modulemanager.  has it been updated to KSP 1.3?

**Comment by scorpianz1525** [comment]
I think so... What version am I looking for?

**Comment by pipakin** [comment]
I'll do a pull to get the latest when I get home from work and build a new binary if people want to keep experimenting.

@Dunbaratu is there someone I should poke about https://github.com/KSP-KOS/KSP_LIB/pull/3 ? Having a failing build is making me a sad panda (also I use github for work so I'm perhaps more paranoid about failed builds than most).

**Comment by Dunbaratu** [comment]
I should have some free time late at night tonight to have a look at it.

File: src/kOS/Communication/kOSConnectivityParameters.cs [hunk 1]
```diff
@@ -64,6 +64,8 @@ public override bool HasPresets
             }
         }
 
+        public override string DisplaySection => "kOS";
+
```
Comment (Dunbaratu) [inline src/kOS/Communication/kOSConnectivityParameters.cs]: This syntax was not understood when I tried to compile this in monodevelop.

I had to change it to the more explicit way of making a property:
```
        public override string DisplaySection { get {return "kOS";} }
```

**Comment by Dunbaratu** [comment]
I've made a pre-release candidate for this public in the reddit forum.  I'll wait a day or two before making the release official, to see if anyone reports any big problems.

In the meantime I'll look into getting Travis's DLLS updated tomorrow.

File: src/kOS/Communication/kOSConnectivityParameters.cs (hunk 1 above)
Comment (DallasEpperson) [inline src/kOS/Communication/kOSConnectivityParameters.cs]: That's a new feature in C#6, expression bodied property. If I'm not mistaken you need mono version 4 or higher to compile it correctly via monodevelop.

Comment (Dunbaratu) [inline src/kOS/Communication/kOSConnectivityParameters.cs]: I have to stick with the version of Monodevelop that ships with the version of Unity that KSP is running on, or else I can't get the debug version to work.

**Pull request merged**